### PR TITLE
Moving cumsum to statistics

### PR DESCRIPTION
**Pull request by justwagle**
This pull request is an attempt to move cumsum primitive to the statistics plugin.

### CODE REVIEW

File: phylanx/plugins/statistics/statistics_base_impl.hpp [hunk 1]
```diff
@@ -481,10 +481,10 @@ namespace phylanx { namespace execution_tree { namespace primitives
         switch (a_dims)
         {
         case 0:
-            return statistics0d(std::move(arg), axis, keepdims, initial_value);
+            return derived().statistics0d(std::move(arg), axis, keepdims, initial_value);
 
         case 1:
-            return statistics1d(std::move(arg), axis, keepdims, initial_value);
+            return derived().statistics1d(std::move(arg), axis, keepdims, initial_value);
```
Comment (hkaiser) [inline phylanx/plugins/statistics/statistics_base_impl.hpp]: Could you add the `derived().` to the Tensor case as well, please?

File: src/plugins/statistics/cumsum_operation.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+        for (std::size_t row = 0; row != m.rows(); ++row)
+        {
+            hpx::parallel::inclusive_scan(hpx::parallel::execution::par,
+                m.begin(row), m.end(row), result.begin(row));
+        }
+
+        return primitive_argument_type{std::move(result)};
+    }
+}}}
```
Comment (hkaiser) [inline src/plugins/statistics/cumsum_operation.cpp]: IIUC, you have not used the base class implementation at all. Is that intended?

File: tests/unit/plugins/statistics/CMakeLists.txt [hunk 3]
```diff
@@ -4,7 +4,8 @@
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 set(tests
-	logsumexp_operation
+    cumsum_operation
+	  logsumexp_operation
```
Comment (hkaiser) [inline tests/unit/plugins/statistics/CMakeLists.txt]: Please change the TAB to spaces.

**Comment by justwagle** [comment]
As we have decided that cumsum primitive may not be best suited for statistics plugin, I will be closing the pull request.

**Pull request closed without merging**